src/cobra/core/reaction.py

```diff
@@ -1,20 +1,33 @@
-# -*- coding: utf-8 -*-
-
 """Define the Reaction class."""
 
-from __future__ import absolute_import, print_function
-
 import hashlib
 import re
 from collections import defaultdict
 from copy import copy, deepcopy
 from functools import partial
 from math import isinf
 from operator import attrgetter
-from typing import FrozenSet
+from typing import (
+    TYPE_CHECKING,
+    AnyStr,
+    Dict,
+    FrozenSet,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
 from warnings import warn
 
-from future.utils import raise_from, raise_with_traceback
+
+if TYPE_CHECKING:
+    from optlang.interface import Variable
+    from cobra import Model, Solution
+    from cobra.summary import ReactionSummary
+    from pandas import DataFrame
```

Comment:
Please use `import pandas as pd` and then refer `DataFrame` as `pd.DataFrame` so as to maintain the standard used by `pandas` community and also make it easy for us to understand foreign imports.